fc/protocol.py

```diff
@@ -445,53 +439,30 @@ def outputs_and_plots(self, errors, verbose=True, write_out=True):
         # Plots
         if write_out:  # suppress plotting when performing fitting
             start = time.time()
-            for plot in self.plots:
-                with errors:
+            with errors:
```

Comment:
I think doing this the other way around means you try each plot, even if earlier ones had errors. Which is what we want.